app/src/main/java/com/github/orkest/ViewModel/auth/AuthViewModel.kt

```diff
@@ -90,26 +86,21 @@ open class AuthViewModel: ViewModel() {
         val firstLetter = username.value.text[0].uppercase()
         val path = "user/user-$firstLetter/users"
 
-
         //Checks if the database already contains a user with the same username
         db.collection(path)
-            .document(user.username).get()
-            .addOnSuccessListener {
+            .document(user.username).get().addOnSuccessListener {
                 if (it.data != null) {
                     println(it)
                     future.complete(false)
                 } else {
                     //If no user with the same username was found, add the user to the database
-                    pushUser(path)
-                        .addOnSuccessListener { future.complete(true) }
+                    pushUser(path).addOnSuccessListener { future.complete(true) }
                 }
-            }
-            //Propagates the exception in case of another exception
+            } //Propagates the exception in case of another exception
             .addOnFailureListener{
                 future.completeExceptionally(
-                    Exception("Sorry, something went wrong ... Please check your Internet connection"))
+                Exception("Sorry, something went wrong ... Please check your Internet connection"))
             }
-
         return future
     }
```

Comment:
I thought you've reduced the number of lines by improving the function, maybe be a bit more explicit in the commit message

Reply:
Okay will do next time thanks!